mission7/schema.sql

```diff
@@ -0,0 +1,173 @@
+CREATE OR REPLACE FUNCTION update_updated_at_column()
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.updated_at = NOW();
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TABLE users (
+  id SERIAL PRIMARY KEY,
+  email VARCHAR(50) UNIQUE NOT NULL,
+  nickname VARCHAR(255) NOT NULL,
+  password VARCHAR(100) NOT NULL,
+  image VARCHAR(255),
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+CREATE TRIGGER trg_users_updated_at
+BEFORE UPDATE ON users
+FOR EACH ROW
+EXECUTE FUNCTION update_updated_at_column();
+
+CREATE TABLE products (
+  id SERIAL PRIMARY KEY,
+  user_id INT NOT NULL,
+  name VARCHAR(50) NOT NULL,
+  description VARCHAR(255),
+  price INT NOT NULL CHECK (price >= 0),
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+  CONSTRAINT fk_products_user FOREIGN KEY (user_id)
+    REFERENCES users(id) ON DELETE CASCADE
+);
+
+CREATE INDEX idx_products_user_id ON products(user_id);
+CREATE INDEX idx_products_created_at ON products(created_at);
+
```

Comment:
피그마 디자인이나, 실제 서비스들에서 있을 법한,
자주 사용될 법한 패턴을 추론할 수 있어야해요.

예를 들면, 내 상품 목록. 같은 상황이겠죠

```suggestion
CREATE INDEX idx_products_user_created_id
  ON products(user_id, created_at DESC, id DESC);
```